scripts/Muon/GUI/Common/grouping_tab_widget/grouping_tab_widget_model.py

```diff
@@ -0,0 +1,124 @@
+from __future__ import (absolute_import, division, print_function)
+
+from collections import OrderedDict
+
+from Muon.GUI.Common.muon_pair import MuonPair
+from Muon.GUI.Common.muon_group import MuonGroup
+from Muon.GUI.Common.muon_data_context import MuonDataContext, construct_empty_group, construct_empty_pair
+
+
+class GroupingTabModel(object):
+    """
+    The model for the grouping tab should be shared between all widgets of the tab.
+    It keeps a record of the groups and pairs defined for the current instance of the interface.
+
+    pairs and groups should be of type MuonGroup and MuonPair respectively.
+    """
+
+    def __init__(self, data=MuonDataContext()):
+        self._data = data
+
+    def get_group_workspace(self, group_name):
+        """
+        Return the workspace associated to group_name, creating one if
+        it doesn't already exist (e.g. if group added to table but no update yet triggered).
+        """
+        try:
+            workspace = self._data.groups[group_name].workspace.workspace
```

Comment:
it seems a bit odd to have `workspace.workspace`

Reply:
I think changing this will require quite a lot of name changing through the code. This will I think be done more easily when there is not a PR backlog so could we mark this down as to change later?